py/fiberassign/run.py

```diff
@@ -0,0 +1,997 @@
+    # with very low priority.
+    #
+    # # Assign safe location to unused fibers (no maximum).  There should
+    # # always be at least one safe location (i.e. "BAD_SKY") for each fiber.
+    # # So after this is run every fiber should be assigned to something.
+    # asgn.assign_unused(TARGET_TYPE_SAFE)
+
+    # Assign sky monitor fibers
+    asgn.assign_unused(TARGET_TYPE_SKY, -1, "ETC")
+
+    gt.stop("run_survey_full calculation")
+    gt.start("run_survey_full write output")
+
+    # Make sure that output directory exists
+    if not os.path.isdir(args.dir):
+        os.makedirs(args.dir)
+
+    # Optionally get GFA targets
+    gfa_targets = None
+    if args.gfafile is not None:
+        gfa_targets = get_gfa_targets(tiles, args.gfafile)
+
+    # Write output
+    write_assignment_fits(tiles, asgn, out_dir=args.dir,
+                          out_prefix=args.prefix, split_dir=args.split,
+                          all_targets=args.write_all_targets,
+                          gfa_targets=gfa_targets, overwrite=args.overwrite)
+
+    gt.stop("run_survey_full write output")
+
+    gt.report()
+
+    return
+
+
+def run_survey_bytile(args):
+    """Run fiber assignment tile-by-tile.
+
+    This uses the previously parsed options to read input data and run through
+    the typical assignment sequence on a single tile before moving on to the
+    next.  It then writes to the outputs to disk.
+
+    Args:
+        args (namespace): The parsed arguments.
+
+    Returns:
+        None
+
+    """
+    gt = GlobalTimers.get()
+    gt.start("run_survey_bytile calculation")
+
+    # Load data
+    hw, tiles, tgs = run_survey_init(args)
+
+    # Create a hierarchical triangle mesh lookup of the targets positions
+    tree = TargetTree(tgs)
+
+    # Compute the targets available to each fiber for each tile.
+    tgsavail = TargetsAvailable(hw, tgs, tiles, tree)
+
+    # Free the tree
+    del tree
+
+    # Compute the fibers on all tiles available for each target and sky
+    favail = FibersAvailable(tgsavail)
+
+    # Create assignment object
+    asgn = Assignment(tgs, tgsavail, favail)
+
+    # We are now going to loop over tiles and assign each one fully before
+    # moving on to the next
+
+    for tile_id in tiles.id:
+
+        # First-pass assignment of science targets
+        asgn.assign_unused(TARGET_TYPE_SCIENCE, -1, "POS", tile_id, tile_id)
+
+        # Assign standards, 10 per petal
+        asgn.assign_unused(TARGET_TYPE_STANDARD, args.standards_per_petal,
+                           "POS", tile_id, tile_id)
+        asgn.assign_force(TARGET_TYPE_STANDARD, args.standards_per_petal,
+                          tile_id, tile_id)
+
+        # Assign sky to unused fibers, up to 40 per petal
+        asgn.assign_unused(TARGET_TYPE_SKY, args.sky_per_petal, "POS",
+                           tile_id, tile_id)
+        asgn.assign_force(TARGET_TYPE_SKY, args.sky_per_petal,
+                          tile_id, tile_id)
+
```

Comment:
After doing these, we should have a final `assign_unused` for both standards and sky to fill in any remaining unused fibers since the previous calls max out at `args.standards_per_petal` and `args.sky_per_petal`.  i.e.
```python
#- I think this first one is a no-op, but would be good to confirm
### asgn.assign_unused(TARGET_TYPE_SCIENCE, -1, "POS", tile_id, tile_id)
asgn.assign_unused(TARGET_TYPE_STANDARD, -1, "POS", tile_id, tile_id)
asgn.assign_unused(TARGET_TYPE_SKY, -1, "POS", tile_id, tile_id)
```

With this addition, this is equivalent to what I tested before for my per-tile serial assignments tests.